src/common/simple_cache.hpp

```diff
@@ -71,6 +73,7 @@ class SimpleLRU {
       return;
     lru.erase(i->second);
     contents.erase(i);
+    total_bytes -= i->second->second.length();
```

Comment:
do the arithmetic before erasing from the containers - iterators are invalidated by erasing them. This might have caused the odd values you were observing.

Reply:
Done.